### PR TITLE
Adding compute machines to vSphere/bare metal UPI environments

### PR DESCRIPTION
**Pull request by codyhoag**
A subset of content from #15773, specifically covering vSphere and bare metal use cases.

### CODE REVIEW

**Comment by codyhoag** [comment]
@jiajliu @morenod can you review these docs for adding compute machines to vSphere and bare metal UPI installations? Does this apply for OCP 4.3-4.6?

**Comment by jiajliu** [comment]
@jinyuma Could u help review vsphere part? Thx!

**Comment by morenod** [comment]
> @jiajliu @morenod can you review these docs for adding compute machines to vSphere and bare metal UPI installations? Does this apply for OCP 4.3-4.6?
> 
> [vSphere preview](https://adding-workers--ocpdocs.netlify.app/openshift-enterprise/latest/machine_management/user_infra/adding-vsphere-compute-user-infra.html)
> [bare metal preview](https://adding-workers--ocpdocs.netlify.app/openshift-enterprise/latest/machine_management/user_infra/adding-bare-metal-compute-user-infra.html)

baremetal steps will be valid from 4.3 to 4.5

starting on 4.6, PXE parameters have changed to live kernel

**Comment by codyhoag** [comment]
>baremetal steps will be valid from 4.3 to 4.5
>
>starting on 4.6, PXE parameters have changed to live kernel

@morenod I did some digging and updated the bare metal PXE params (ed0292a711b2f7c43680d088779580ee932bc426) based on what I found. Can you review and let me know if I'm on the right track and if any additional changes are required? I'll separate this change from 4.3-4.5 after I have verification. Thanks!

**Comment by codyhoag** [comment]
@morenod can you review the updates I added for bare metal live kernel PXE params based on your feedback (3f19ec61714d5f5aff891351d2c7f9d5c1f343cf)? I'll separate this change from 4.3-4.5 after I have verification.

@jinyunma can you review the vSphere portion of this PR?

[bare metal preview](https://adding-workers--ocpdocs.netlify.app/openshift-enterprise/latest/machine_management/user_infra/adding-bare-metal-compute-user-infra.html)
[vSphere preview](https://adding-workers--ocpdocs.netlify.app/openshift-enterprise/latest/machine_management/user_infra/adding-vsphere-compute-user-infra.html)

**Comment by jinyunma** [comment]
During I verified the steps, I have already added some comments above, but seems not consider yet in [vSphere preview](https://adding-workers--ocpdocs.netlify.app/openshift-enterprise/latest/machine_management/user_infra/adding-vsphere-compute-user-infra.html)
Could you please help to check?

**Comment by jinyunma** [comment]
During I verified the steps, I have already added some comments above, but seems not consider yet in [vSphere preview](https://adding-workers--ocpdocs.netlify.app/openshift-enterprise/latest/machine_management/user_infra/adding-vsphere-compute-user-infra.html)
Could you please help to check?

**Comment by codyhoag** [comment]
@jinyunma I don't see any of your comments in this PR. Please let me know if I'm overlooking something.

File: modules/machine-user-infra-machines-pxe.adoc [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+** For PXE:
++
+----
+DEFAULT pxeboot
+TIMEOUT 20
+PROMPT 0
+LABEL pxeboot
+    KERNEL http://<HTTP_server>/rhcos-<version>-installer-live-kernel-<architecture> <1>
+    APPEND ip=dhcp rd.neednet=1 initrd=http://<HTTP_server>/rhcos-<version>-installer-live-initramfs.<architecture>.img console=tty0 console=ttyS0 coreos.inst=yes coreos.inst.install_dev=sda coreos.inst.image_url=http://<HTTP_server>/rhcos-<version>-metal.<architecture>.raw.gz coreos.live.rootfs_url=http://<HTTP_server>/rhcos-<version>-installer-live-rootfs.<architecture>.img coreos.inst.ignition_url=http://<HTTP_server>/worker.ign <2> <3>
```
Comment (morenod) [inline modules/machine-user-infra-machines-pxe.adoc:37]: This is not correct. if using live kernel and initramfs, coreos.inst.install_dev needs to point to the full path (/dev/sda) and coreos.inst.image_url is not required.

Also ip=dhcp, rd.neednet and coreos.inst should be removed, as they are deprecated.

Correct line on 4.6 is:

initrd=http://\<HTTP_server\>/rhcos-\<version\>-installer-live-initramfs.\<architecture\>.img console=ttyS0 console=tty0 coreos.inst.install_dev=/dev/sda coreos.inst.ignition_url=http://\<HTTP_server\>/worker.ign coreos.live.rootfs_url=http://\<HTTP_server\>/rhcos-\<version\>-installer-live-rootfs.\<architecture\>.img

File: modules/machine-user-infra-machines-pxe.adoc [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+For example, to use DHCP on a NIC that is named `eno1`, set `ip=eno1:dhcp`.
+<3> Specify locations of the {op-system} files that you uploaded to your
+HTTP server. The `initrd` parameter value is the location of the live `initramfs` file,
+the `coreos.inst.image_url` parameter value is the location of the compressed
+metal RAW image, the `coreos.live.rootfs_url` parameter value is the location of the live `rootfs` file, and the `coreos.inst.ignition_url` parameter value is the location of the worker Ignition config file.
+
+** For iPXE:
++
+----
+kernel http://<HTTP_server>/rhcos-<version>-installer-kernel-<architecture> ip=dhcp rd.neednet=1 initrd=http://<HTTP_server>/rhcos-<version>-installer-initramfs.<architecture>.img console=tty0 console=ttyS0 coreos.inst=yes coreos.inst.install_dev=sda coreos.inst.image_url=http://<HTTP_server>/rhcos-<version>-metal.<arhcitectutre>.raw.gz coreos.inst.ignition_url=http://<HTTP_server>/worker.ign <1> <2>
```
Comment (morenod) [inline modules/machine-user-infra-machines-pxe.adoc:51]: Same than PXE, correct line is:

kernel http://\<HTTP_server\>/rhcos-\<version\>-installer-kernel-\<architecture\> initrd=http://\<HTTP_server\>/rhcos-\<version\>-installer-live-initramfs.\<architecture\>.img console=ttyS0 console=tty0 coreos.inst.install_dev=/dev/sda coreos.inst.ignition_url=http://\<HTTP_server\>/worker.ign coreos.live.rootfs_url=http://\<HTTP_server\>/rhcos-\<version\>-installer-live-rootfs.\<architecture\>.img

**Review by morenod — CHANGES_REQUESTED** [review]

**Comment by codyhoag** [comment]
@morenod @jinyunma thanks for your feedback! Can you verify the updated content?

**bare metal**
- Changes: b79c70f0fe2bdf1bc608d949a83a0c490a7a3a00
- Preview: https://adding-workers--ocpdocs.netlify.app/openshift-enterprise/latest/machine_management/user_infra/adding-bare-metal-compute-user-infra.html

**vSphere**
- Changes: ce9e0de85bb692ed88e171247e1e4a81bb1edebe
- Preview: https://adding-workers--ocpdocs.netlify.app/openshift-enterprise/latest/machine_management/user_infra/adding-vsphere-compute-user-infra.html

**Comment by jinyunma** [comment]
LGTM for vsphere part.

**Review by morenod — APPROVED** [review]
LGTM

**Comment by codyhoag** [comment]
Since this includes 4.6-only edits, I will separate the PRs for 4.3-4.5. The 4.5 PR (which will match 4.3-4.4) is #25361. This content has also been verified by QE.

**Review by bobfuru — COMMENTED** [review]
This is great work! A couple of prereqs aren't rendering in the preview build, might just be a caching issue. A few other suggestions but overall, LGTM!

Technically, I believe the `Milestone` label should be `Future release` since this is for the as-yet-unreleased 4.6.

File: machine_management/user_infra/adding-bare-metal-compute-user-infra.adoc [hunk 3]
```diff
@@ -0,0 +1,29 @@
+[id="adding-bare-metal-compute-user-infra"]
+= Adding compute machines to bare metal
+include::modules/common-attributes.adoc[]
+:context: adding-bare-metal-compute-user-infra
+
+toc::[]
+
+You can add more compute machines to your {product-title} cluster on bare metal.
+
+== Prerequisites
```
[the remark this answers is not in the document]
Comment (codyhoag) [reply machine_management/user_infra/adding-bare-metal-compute-user-infra.adoc:10]: Yeah, that was a last minute change I made to follow our updated guidelines for assembly prereq headings. It appears the bot didn't want to update the preview :disappointed: . Just checked locally and it appears as expected: http://file.rdu.redhat.com/~choag/091020/adding-workers/machine_management/user_infra/adding-bare-metal-compute-user-infra.html

File: machine_management/user_infra/adding-vsphere-compute-user-infra.adoc [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+You can add more compute machines to your {product-title} cluster on VMware
+vSphere.
+
+== Prerequisites
+
+* You xref:../../installing/installing_vsphere/installing-vsphere.adoc#installing-vsphere[installed a cluster on vSphere].
+
+include::modules/machine-vsphere-machines.adoc[leveloffset=+1]
+
+include::modules/installation-approve-csrs.adoc[leveloffset=+1]
```
[the remark this answers is not in the document]
Comment (codyhoag) [reply machine_management/user_infra/adding-vsphere-compute-user-infra.adoc:17]: Confirmed locally: http://file.rdu.redhat.com/~choag/091020/adding-workers/machine_management/user_infra/adding-vsphere-compute-user-infra.html

File: modules/installation-approve-csrs.adoc [hunk 5]
```diff
@@ -21,7 +23,7 @@ endif::[]
 [id="installation-approve-csrs_{context}"]
 = Approving the CSRs for your machines
 
-When you add machines to a cluster, two pending certificates signing request
+When you add machines to a cluster, two pending certificate signing requests
```
[the remark this answers is not in the document]
Comment (codyhoag) [reply modules/installation-approve-csrs.adoc:26]: This term appears to be used quite often outside of OCP and the hyphen is not used in those cases.

**Pull request merged**